test/linear_combination.jl

```diff
@@ -60,7 +60,7 @@
             @testset "dims = :" begin
                 M = [1 1; 2 2; 3 5]
                 lc = LinearCombination([1, -1, 1])
-                @test_throws ArgumentError FeatureTransforms.apply(M, lc; dims=:)
+                @test_deprecated FeatureTransforms.apply(M, lc; dims=:)
```

Comment:
We have removed the default `dims=1` for LinearCombination but added a deprecation warning so it's non-breaking for now.